src/coreComponents/physicsSolvers/wavePropagation/ElasticVTIWaveEquationSEM.cpp

```diff
@@ -0,0 +1,773 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+
+/**
+ * @file ElasticVTIWaveEquationSEM.cpp
+ */
+
+#include "ElasticVTIWaveEquationSEM.hpp"
+#include "ElasticVTIWaveEquationSEMKernel.hpp"
+
+#include "fieldSpecification/FieldSpecificationManager.hpp"
+#include "finiteElement/FiniteElementDiscretization.hpp"
+#include "mainInterface/ProblemManager.hpp"
+#include "mesh/ElementType.hpp"
+#include "mesh/mpiCommunications/CommunicationTools.hpp"
+#include "WaveSolverUtils.hpp"
+#include "ElasticTimeSchemeSEMKernel.hpp"
+#include "ElasticMatricesSEMKernel.hpp"
+#include "PrecomputeSourcesAndReceiversKernel.hpp"
+
+namespace geos
+{
+
+using namespace dataRepository;
+using namespace fields;
+
+ElasticVTIWaveEquationSEM::ElasticVTIWaveEquationSEM( const std::string & name,
```

Comment:
Have this class inherit from ElasticWaveEquationSEM. You can avoid:
- almost all the registerWrapper calls in the initializer (just add the new ones)
- most of the registerDataOnMesh method (jsut add register for the new fields)
- essentially everythig else in this class.

To do this, you can template the ElasticWaveEquation calss as 
```
template< typename KERNELS = ElasticWaveEquationSEMKernels >
class ElasticWaveEquation :: public WaveSolverBase { bla bla bla }
```
and then
```
class ElasticWaveVTIEquation :: public ElasticWaveEquation<ElasticVTIWaveEquationKernels > { bla bla bla }
```